example-app/SantokuApp/src/bases/core/utils/usePrevious.ts

```diff
@@ -3,7 +3,7 @@ import {useEffect, useRef} from 'react';
 /**
  * This React hook provides the previous value. (To be precise, it returns the previous reference.)
  *
- * See [Hooks FAQ - How to get the previous props or state?](https://reactjs.org/docs/hooks-faq.html#how-to-get-the-previous-props-or-state).
+ * See [Hooks FAQ - How to get the previous props or state?](https://legacy.reactjs.org/docs/hooks-faq.html#how-to-get-the-previous-props-or-state).
```

Comment:
legacyからも`usePrevious`に関する記載は削除されているので、新しいドキュメントに移行されることもないと思います。
なので、このリンクは削除して、legacyのドキュメントに記載されている注意事項などを記載しておいた方が良いと思いました。

>We have previously suggested a custom Hook called usePrevious to hold the previous value. However, we’ve found that most use cases fall into the two patterns described above. If your use case is different, you can [hold a value in a ref](https://legacy.reactjs.org/docs/hooks-faq.html#is-there-something-like-instance-variables) and manually update it when needed. Avoid reading and updating refs during rendering because this makes your component’s behavior difficult to predict and understand.
以前、usePreviousというカスタムHookで前の値を保持することを提案しました。しかし、ほとんどのユースケースは上記の2つのパターンに当てはまることがわかりました。ユースケースが異なる場合は、refに値を保持し、必要なときに手動で更新することができます。レンダリング中にrefを読み込んだり更新したりするのは、コンポーネントの動作の予測や理解が困難になるため避けましょう。

Reply:
注意事項とドキュメントのリンク両方記載しました。
対応Commit: 2b7a1b7cbb8c2007fb8474bc2075a14bf914cd7f

Reply:
以下の理由からlegacyページへのリンクを残しておいたほうが良さそうです

- コピーしているので出典を示すため
  - 引用であることがわかるようにしたほうが良さそうです（先頭に `> `追加？）
    - > We have previously suggested a custom Hook called usePrevious to hold the previous value.
    - > 以前の値を保持するために usePrevious というカスタム フックを提案しました。
- 以下の文章が引用された部分だけだと理解できない
  - > However, we’ve found that most use cases fall into the two patterns described above.
  - > しかし、ほとんどのユースケースは上記の2つのパターンに当てはまることがわかりました。

Reply:
この `usePrevious` は非推奨としたいんですかね？
ならば、 `We have previously suggested...`の前にその旨を記載しておくと良さそうです

Reply:
非推奨とまではいかないようには読み取ってましたね。
- 基本は、`usePrevious`を使用しないパターンで実装するべき
- どうしても`Ref`を使う必要がある場面だけ`usePrevious`を使おうね

Reply:
そうですね。
乱用はしないようにって程度ですね。

↑の2点を記載しておくと良さそうです

Reply:
ありがとうございます。対応しました。
対応Commit: 68a2fa1fcb5433cb8777747a9a4623a8e36adf23